### PR TITLE
Creating a user profile

### LINKED ISSUE #51: Adding a user model
Use the built-in Django authentication system for the basic fields and an additional table for more info

### PR DESCRIPTION
**Pull request by Leeadar**
- Extends the built-in Django User model
- User has a field of a profile image
- Creating a media folder
- Make our static files accessible through our URL patterns

**For reviewing PR through the admin interface:**
1. Add a new profile
2. Your username, image and email appears at http://127.0.0.1:8000/profile/
Try to create another profile without uploading an image, and the default image appears in the profile page.

image

closes #51

### CODE REVIEW

File: Pipfile [hunk 1]
```diff
@@ -9,6 +9,7 @@ flake8 = "*"
 [packages]
 django = "*"
 django-crispy-forms = "*"
+pillow = "*"
```
Comment (guy9050) [inline Pipfile:12]: where do you use this package?

File: users/models.py [hunk 2]
```diff
@@ -1,3 +1,11 @@
...
 
-# Create your models here.
+
+class Profile(models.Model):
+    user = models.OneToOneField(User, on_delete=models.CASCADE)
+    image = models.ImageField(default='default.jpg', upload_to='profile_pics')
+    dob = models.DateField()
+
+    def __str__(self):
+        return f'{self.user.username} Profile'
```
Comment (guy9050) [inline users/models.py:11]: please add a blank line at the end

Comment (Leeadar) [reply users/models.py:11]: There is a blank line in line 12. 
Do you mean to add another one (line 13) ?

File: Pipfile (hunk 1 above)
Comment (Leeadar) [reply Pipfile:12]: I am using the package Inside models.py, in the 'ImageField'

**Review by maayanhd — CHANGES_REQUESTED** [review]
Please erase the content in the admin.py so there won't be conflicts and error when the PR related to issue #61 will be merged

File: users/models.py (hunk 2 above)
Comment (guy9050) [reply users/models.py:11]: no only line 12, but I don't see a blank line there :) https://github.com/beyond-io/picATrip/blob/df9933d0d7c3f7c1e5957314c765280e9dc9cb9d/users/models.py

**Comment by Leeadar** [comment]
@maayanhd I erased the content in the admin.py

Regarding the profile page, the default picture will work after you will create a profile. In the future it the profile will be created in the registration page, but for this PR, the profile can be created only in the Django admin UI.

File: users/models.py (hunk 2 above)
Comment (guy9050) [reply users/models.py:11]: @Leeadar

Comment (Leeadar) [reply users/models.py:11]: I'm sorry I didn't send my answer, do you see it now? :)

Comment (guy9050) [reply users/models.py:11]: yes but it's the notepad++ that added a fake blank file at the end, git doesn't do these things:)
also I opened it in vim to verify:
image

Comment (Leeadar) [reply users/models.py:11]: I tried to add a new line (and then delete line 13), is it work now ?

**Review by maayanhd — APPROVED** [review]
consider using this field instead of DateField, it might be useful later, but it will require to add to the Bootstrap file adding the installation command I believe
https://pypi.org/project/django-birthday/#:~:text=django%2Dbirthday%20provides%20a%20birthday,is%20a%20subclass%20of%20django.&text=It%20also%20internally%20adds%20a,extra%20functionality%20exposed%20by%20birthday.

File: users/models.py (hunk 2 above)
Comment (guy9050) [reply users/models.py:11]: no.. it shows no change in this file

Comment (Leeadar) [reply users/models.py:11]: How can I open the file with vim like you did? I searched and didn't find how can I do that.
Maybe I can add the blank line in the vim editor... (I tried also with notepad, and it also shows me that line 12 is blank)

Comment (guy9050) [reply users/models.py:11]: It's a unix based text editor so you have to run on another OS.. but I think if you open it in any IDE (visual studio, pycharm.. etc) you won't see that empty line

Comment (Leeadar) [reply users/models.py:11]: With Visual Studio code is the same result..
 
image

Comment (guy9050) [reply users/models.py:11]: hmm so maybr you forgot to add/commit that file when you pushed to git? if you run `git status` do you see it?

Comment (Leeadar) [reply users/models.py:11]: No.. its says 'nothing to commit, working tree clean' (in user-model branch)

**Comment by Yarboa** [comment]
> Please erase the content in the admin.py so there won't be conflicts and error when the PR related to issue #61 will be merged.
> Also any idea why the default picture doesn't work for me? -
> image
> maybe look here (https://stackoverflow.com/questions/15140483/django-imagefield-setting-a-fixed-width-and-height) for resizing the picture next versions

I still have it

File: users/models.py (hunk 2 above)
Comment (Yarboa) [reply users/models.py:11]: You could look inside vagrant, /usr/bin/vi
It look ok, for me

**Comment by maayanhd** [comment]
@Yarboa unless you add a profile and connect it through your admin you won't see a default picture, this what I understood from @Leeadar anyway

File: users/models.py (hunk 2 above)
Comment (Leeadar) [reply users/models.py:11]: Following my meet with Guy we decided to keep this file as before for now. Maybe there is a problem with Flake8 that indicate an extra new line

**Comment by Leeadar** [comment]
@Yarboa This is the result of the test data for creating new profiles:

image

**Comment by Yarboa** [comment]
> @Yarboa unless you add a profile and connect it through your admin to the user you are connected with to the site, you won't see a default picture, this what I understood from @Leeadar anyway

Ack

File: users/migrations/0001_initial.py [hunk 3]
```diff
@@ -0,0 +1,26 @@
+# Generated by Django 3.2 on 2021-04-08 16:22
```
[the remark this answers is not in the document]
Comment (Yarboa) [reply users/migrations/0001_initial.py:1]: I see 0002_test_data.py have added

**Review by Yarboa — APPROVED** [review]

**Pull request merged**